lib/src/string_scanner.dart

```diff
@@ -120,6 +139,45 @@ class StringScanner {
     _fail(name);
   }
 
+  /// Consumes a single Unicode code unit and returns it.
+  ///
+  /// This works like [readChar], except that it automatically handles UTF-16
+  /// surrogate pairs. Specifically, if the next character is a high surrogate,
+  /// it consumes it as well as the following character and returns the resolved
+  /// Unicode code point.
+  ///
+  /// If the scanner is looking at an unpaired surrogate, this returns the high
```

Comment:
"this" -> "this call"
I'd probably rephrase as:

> If next two characters are not a surrogate pair, the next single code point is returned as-is, even if the codepoint is an unpaired surrogate.


(Generally, never have an unqualified "this", or "it" or "that". The sentence always gets more readable if you write the unqualified clause out.)